scipy/signal/_peak_finding.py

```diff
@@ -439,6 +439,14 @@ def peak_widths(x, peaks, rel_height=0.5, prominence_data=None, wlen=None):
         Interpolated positions of left and right intersection points of a
         horizontal line at the respective evaluation height.
 
+    Raises
+    ------
+    ValueError
+        If `prominence_data` is supplied but doesn't satisfy the condition
+        ``0 <= left_base <= peak <= right_base < x.shape[0]`` for each peak,
+        has the wrong dtype, is not C-contiguous or does not have the same
```

Comment:
Why do you need it to be C-contiguous?

Reply:
If I declare the arrays `prominence_data` as C-contiguous in `_peak_widths`'s signature I get noticeable speed gains. 

The argument `prominece_data` is mainly there so that I don't need to calculate prominences twice for the case `find_peaks(..., prominence=..., width=...)`. I didn't want to add more input validation, tests and casting for the (in my opinion) rare case that the user really wants to supply these arrays himself.